src/tests/ftest/cart/test_group_np_srv.c

```diff
@@ -43,17 +43,14 @@ swim_crt_event_cb(d_rank_t rank, enum crt_event_source src,
 		"rank = %d, "
 		"crt_event_source = %d, "
 		"crt_event_type = %d\n",
-		 rank, src, type);
+		rank, src, type);
 
-	/* convert integer to string */
-	snprintf(type_to_a, 2, "%d", type);
+	swim_state_str[0] = type + '0';
+	swim_state_str[1] = 0;
 
-	if (swim_status_by_rank[rank] == NULL) {
-		swim_status_by_rank[rank] = (char *)malloc(max * sizeof(char));
-		strcpy(swim_status_by_rank[rank], "\0");
-	}
-
-	strcat(swim_status_by_rank[rank], type_to_a);
+	maxlen = MAX_SWIM_STATUSES - strlen(swim_state_str);
```

Comment:
Isn't strlen(swim_state_str) always 1 ?

Reply:
For now, yes -- it only gathers one swim state at a time.